packages/nimble-components/src/table-column/number-text/tests/table-column-number-text-matrix.stories.ts

```diff
@@ -82,3 +100,87 @@ tableColumnNumberTextThemeMatrix.play = async (): Promise<void> => {
         )
     );
 };
+
+interface NumberTextColumnTableArgs extends SharedTableArgs {
+    unit: string;
+}
+
+interface KeyedObject {
+    [key: string]: number;
+}
+
+const largeDataSets: KeyedObject[][] = [[], []];
+let largeDataSetIndex = 0;
+const rowsLargeData = 30;
+const columnsLargeData = 20;
+let updateDataSetFunction: (() => Promise<void>) | null;
+
+for (let set = 0; set < 2; set++) {
+    for (let i = 0; i < rowsLargeData; i++) {
+        const row: KeyedObject = {};
+        for (let j = 0; j < columnsLargeData; j++) {
+            row[`col${j}`] = i * j + j + set * 100;
+        }
+        largeDataSets[set]!.push(row);
+    }
+}
+
+export const largeTable: StoryObj<NumberTextColumnTableArgs> = {
```

Comment:
I had requested the large table so we know that it did not regress the performance of number columns without units compared to those with, i.e. that unit support was not adding a general tax to all number rendering since every number does go through the UnitFormat system.

It accomplishes that but after playing with it I'd like to see more configuration for the different UnitFormats, for it to use numeric values that cause different scale factors to actually be rendered (currently it is never rendering anything other than the base scale factor for the units selected), for more column types than just numeric, and I'd like to see a stress test of the limits of update performance for profiling (better understand where the costs of table updates actually are).

I'm going to split this out into a separate PR to play with organizing it a bit for those goals.

Reply:
lol [relevant question](https://teams.microsoft.com/l/message/19:0c59ac4a4cc1438894cb42f4b21703f9@thread.skype/1705423311927?tenantId=87ba1f9a-44cd-43a6-b008-6fdb45a5204e&groupId=41626d4a-3f1f-49e2-abdc-f590be4a329d&parentMessageId=1705423311927&teamName=ASW%20SystemLink&channelName=UI%20Working%20Group&createdTime=1705423311927) in teams. Yea while we did perf tests in prototyping, we haven't really validated it now that we are further along with more complex columns and with fairly complex rendering implemented with column sizing, etc. Would be good to have confidence working with clients